airbyte-cdk/python/airbyte_cdk/sources/file_based/config/file_based_stream_config.py

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import codecs
+from typing import Any, List, Mapping, Optional, Union
+
+from airbyte_cdk.models import ConfiguredAirbyteCatalog
+from pydantic import BaseModel, root_validator, validator
+
+PrimaryKeyType = Optional[Union[str, List[str], List[List[str]]]]
+
+
+class CsvFormat(BaseModel):
+    filetype: str
+    delimiter: str = ","
+    quote_char: str = '"'
+    escape_char: Optional[str]
+    encoding: Optional[str] = "utf8"
+    double_quote: bool
+    # Noting that the existing S3 connector had a config option newlines_in_values. This was only supported by pyarrow and not
+    # the Python csv package. It has a little adoption, but long term we should ideally phase this out because of the drawbacks
+    # of using pyarrow
+
+    @validator("delimiter")
+    def validate_delimiter(cls, v):
+        if len(v) != 1:
+            raise ValueError("delimiter should only be one character")
+        if v in {"\r", "\n"}:
+            raise ValueError(f"delimiter cannot be {v}")
+        return v
+
+    @validator("quote_char")
+    def validate_quote_char(cls, v):
+        if len(v) != 1:
+            raise ValueError("quote_char should only be one character")
+        return v
+
+    @validator("escape_char")
+    def validate_escape_char(cls, v):
+        if len(v) != 1:
+            raise ValueError("escape_char should only be one character")
+        return v
+
+    @validator("encoding")
+    def validate_encoding(cls, v):
+        try:
+            codecs.lookup(v)
+        except LookupError:
+            raise ValueError(f"invalid encoding format: {v}")
+        return v
+
+
+class FileBasedStreamConfig(BaseModel):
```

Comment:
I had to move the FileBasedStreamConfig to its own module because now as part of `parse_records()` and `infer_schema()`, we need to pass in the Config creates a circular dependency since stream.DefaultFileBasedStream → CsvParser → stream.FileBasedStreamConfig. By moving the package from stream -> config the circular dependency is removed